### PR TITLE
show ports in table after deployment / inspect

### PR DESCRIPTION
**Pull request by steiler**
```
INFO[0000] Parsing & checking topology file: linux.clab.yml 
+---+-----------------+--------------+---------------+-------+---------+----------------+----------------------+--------------------------+
| # |      Name       | Container ID |     Image     | Kind  |  State  |  IPv4 Address  |     IPv6 Address     |          Ports           |
+---+-----------------+--------------+---------------+-------+---------+----------------+----------------------+--------------------------+
| 1 | clab-linux01-l1 | d7ddd9653556 | alpine:latest | linux | running | 172.20.20.3/24 | 2001:172:20:20::3/64 | 0.0.0.0:6010/tcp -> 8080 |
|    |                 |              |               |       |         |                |                      | :::6010/tcp -> 8080      |
| 2 | clab-linux01-l2 | 013154efda2b | alpine:latest | linux | running | 172.20.20.2/24 | 2001:172:20:20::2/64 | 0.0.0.0:6009/tcp -> 8080 |
|    |                 |              |               |       |         |                |                      | :::6009/tcp -> 8080      |
+---+-----------------+--------------+---------------+-------+---------+----------------+----------------------+--------------------------+
```
What is not sooo nice is, that the table gets wider and you therefore have to have a wider screen to not have any line breaks within a row.

another option might be to also introduce a port command as docker has, to figure out the ports in a seperate output.

This cmd lists port bindings:

```
❯ docker ps --format "table {{.Names}}\t{{.Ports}}" 
NAMES           PORTS
clab-ports-l1   0.0.0.0:32778->80/tcp, :::32778->80/tcp
clab-srl-srl
```

### CODE REVIEW

**Comment by hellt** [comment]
@steiler yeah, I agree having a long table is not great.
Maybe we just put it in the docs that users could use `docker ports` command? Or indeed `clab ports list`

**Comment by steiler** [comment]
Alright, so the port-bindings field in now part of the auto.tmpl. Hence will be outputted as part of the topology-data.json.
Further did I remove the port field in the output table again.

**Comment by steiler** [comment]
So we have the below now.
I'm not sure if this is maybe already sufficient or if we need another `ports list` command that prints tables.

```
sudo ./containerlab -t /home/mava/projects/containerlab/lab-examples/linux.clab.yml inspect -f json --details
INFO[0000] Parsing & checking topology file: linux.clab.yml
[
  {
    "Names": [
      "clab-linux01-l1"
    ],
    "ID": "5fd2631796d82a4caf220a14577eb4a1ed4244ca53d7ea27857ea56f4cdb0c6e",
    "ShortID": "5fd2631796d8",
    "Image": "alpine:latest",
    "State": "running",
    "Status": "Up About a minute",
    "Labels": {
      "clab-mgmt-net-bridge": "br-6fd285acdfec",
      "clab-node-group": "",
      "clab-node-kind": "linux",
      "clab-node-lab-dir": "/home/mava/projects/containerlab/clab-linux01/l1",
      "clab-node-name": "l1",
      "clab-node-type": "",
      "clab-topo-file": "/home/mava/projects/containerlab/lab-examples/linux.clab.yml",
      "containerlab": "linux01"
    },
    "Pid": 0,
    "NetworkSettings": {
      "IPv4addr": "172.20.20.2",
      "IPv4pLen": 24,
      "IPv4Gw": "172.20.20.1",
      "IPv6addr": "2001:172:20:20::2",
      "IPv6pLen": 64,
      "IPv6Gw": "2001:172:20:20::1"
    },
    "Mounts": [
      {
        "Source": "",
        "Destination": ""
      }
    ],
    "Ports": [
      {
        "host_ip": "0.0.0.0",
        "host_port": 6089,
        "port": 8080,
        "protocol": "tcp"
      },
      {
        "host_ip": "::",
        "host_port": 6089,
        "port": 8080,
        "protocol": "tcp"
      },
      {
        "host_ip": "0.0.0.0",
        "host_port": 6088,
        "port": 8081,
        "protocol": "tcp"
      },
      {
        "host_ip": "::",
        "host_port": 6088,
        "port": 8081,
        "protocol": "tcp"
      },
      {
        "host_ip": "0.0.0.0",
        "host_port": 6087,
        "port": 8082,
        "protocol": "tcp"
      },
      {
        "host_ip": "::",
        "host_port": 6087,
        "port": 8082,
        "protocol": "tcp"
      }
    ]
  },
  {
    "Names": [
      "clab-linux01-l2"
    ],
    "ID": "d06fd4ca34fcd966f249d951aefe936f3fdacc346f3bc6a88d9f8584efd72dbf",
    "ShortID": "d06fd4ca34fc",
    "Image": "alpine:latest",
    "State": "running",
    "Status": "Up About a minute",
    "Labels": {
      "clab-mgmt-net-bridge": "br-6fd285acdfec",
      "clab-node-group": "",
      "clab-node-kind": "linux",
      "clab-node-lab-dir": "/home/mava/projects/containerlab/clab-linux01/l2",
      "clab-node-name": "l2",
      "clab-node-type": "",
      "clab-topo-file": "/home/mava/projects/containerlab/lab-examples/linux.clab.yml",
      "containerlab": "linux01"
    },
    "Pid": 0,
    "NetworkSettings": {
      "IPv4addr": "172.20.20.3",
      "IPv4pLen": 24,
      "IPv4Gw": "172.20.20.1",
      "IPv6addr": "2001:172:20:20::3",
      "IPv6pLen": 64,
      "IPv6Gw": "2001:172:20:20::1"
    },
    "Mounts": [
      {
        "Source": "",
        "Destination": ""
      }
    ],
    "Ports": [
      {
        "host_ip": "0.0.0.0",
        "host_port": 6090,
        "port": 8080,
        "protocol": "tcp"
      },
      {
        "host_ip": "::",
        "host_port": 6090,
        "port": 8080,
        "protocol": "tcp"
      }
    ]
  }
]
```

File: types/types.go [hunk 1]
```diff
@@ -129,7 +129,7 @@ type NodeConfig struct {
 	// Bind mounts strings (src:dest:options).
 	Binds []string `json:"binds,omitempty"`
 	// PortBindings define the bindings between the container ports and host ports
-	PortBindings nat.PortMap `json:"portbindings,omitempty"`
+	PortBindings nat.PortMap `json:"-"`
```
Comment (hellt) [inline types/types.go:132]: was this removed for testing?

Comment (hellt) [reply types/types.go:132]: or you removed it because portbindings is now substituted by `port-bindings` which is calculated based on runtime state?

Comment (steiler) [reply types/types.go:132]: Yes the latter was my thinking. The assumption is, that you're interested in the actual running state, which is a concret port per mapping and the possible range is pretty much of no interest...

Comment (steiler) [reply types/types.go:132]: https://github.com/srl-labs/containerlab/pull/1430/files#diff-bb9cd8f6a91720041aaf633ac5fbaf759782c31c70128e050bc1ef0d027dd755R46

File: clab/export.go [hunk 2]
```diff
@@ -70,6 +70,11 @@ func (c *CLab) exportTopologyDataWithTemplate(w io.Writer, p string) error {
 
 	for _, n := range c.Nodes {
 		e.NodeConfigs[n.Config().ShortName] = n.Config()
+		gc, err := n.GetContainers(ctx)
+		if err != nil {
+			return err
+		}
```
Comment (hellt) [inline clab/export.go:76]: @steiler works nicely, this piece is the only thing that I am hesitant about.
Doing an API Get for all nodes just to fetch port bindings is not optimal:

1. not all nodes have port bindings defined
2. and even if they have them, maybe we can just update the binding information once we created the node? I wonder if we already receive the port binding info as a result of a container create?

Comment (steiler) [reply clab/export.go:76]: since you've now made it part of NodeConfig, we should be able to get this info earlier on as part of the deployment, yes.

**Comment by hellt** [comment]
@steiler I think I messed something up during the cleanup, or maybe it was not working, but when I do `clab ins -a -f json --details` I don't see Ports =(

**Comment by hellt** [comment]
Thanks @steiler 
This looks good now. Let's ship it

**Pull request merged**